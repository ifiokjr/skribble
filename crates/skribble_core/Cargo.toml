[package]
name = "skribble_core"
version = { workspace = true }
authors = { workspace = true }
categories = { workspace = true }
documentation = "https://docs.rs/skribble_core"
edition = { workspace = true }
homepage = { workspace = true }
include = ["src/**/*.rs", "Cargo.toml", "readme.md"]
keywords = ["build", "atomic", "css"]
license = { workspace = true }
readme = "readme.md"
repository = { workspace = true }
rust-version = { workspace = true }
description = "Core features for the skribble atomic css library"

[dependencies]
derivative = { workspace = true }
derive_more = { workspace = true }
doc-comment = { workspace = true }
globset = { workspace = true }
heck = { workspace = true }
indent_write = { workspace = true }
indexmap = { features = ["serde"], workspace = true }
lazy_static = { workspace = true }
lightningcss = { workspace = true }
readonly = { workspace = true }
regex = { workspace = true }
serde = { workspace = true }
serde_json = { workspace = true }
skribble_color = { workspace = true, features = ["serde"] }
thiserror = { workspace = true }
typed-builder = { workspace = true }
vfs = { workspace = true }

[dev-dependencies]
insta = { features = ["json"], workspace = true }
