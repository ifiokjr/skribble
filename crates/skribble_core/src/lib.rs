#![deny(clippy::all)]

pub use error::*;

mod error;
