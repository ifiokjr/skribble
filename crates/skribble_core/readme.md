# skribble_core

> core library for skribble

<br />

[![Crate][crate-image]][crate-link] [![Docs][docs-image]][docs-link] [![Status][ci-status-image]][ci-status-link] [![Unlicense][unlicense-image]][unlicense-link]

## Installation

```toml
[dependencies]
skribble_core = "0.0.0"
```

### Usage

```rust
use skribble_core::*;
```

[crate-image]: https://img.shields.io/crates/v/skribble_core.svg
[crate-link]: https://crates.io/crates/skribble_core
[docs-image]: https://docs.rs/skribble_core/badge.svg
[docs-link]: https://docs.rs/skribble_core
[ci-status-image]: https://github.com/ifiokjr/skribble/workflows/ci/badge.svg
[ci-status-link]: https://github.com/ifiokjr/skribble/actions?query=workflow:ci
[unlicense-image]: https://img.shields.io/badge/license-Unlicence-blue.svg
[unlicense-link]: https://opensource.org/license/unlicense
