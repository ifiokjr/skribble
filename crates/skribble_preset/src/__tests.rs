use skribble_core::*;

use super::*;

#[test]
fn default_can_be_added_to_runner() -> AnyEmptyResult {
  let plugin = PresetDefault::builder().build();
  let config: StyleConfig = StyleConfig::builder()
    .plugins(vec![PluginContainer::from(plugin)])
    .build();

  let mut runner = SkribbleRunner::new(config);
  let runner_config = runner.initialize()?;
  insta::assert_debug_snapshot!(runner_config);

  Ok(())
}

#[test]
fn contained() -> AnyEmptyResult {
  let plugin = PresetDefault::builder().build();
  let config: StyleConfig = StyleConfig::builder()
    .plugins(vec![PluginContainer::from(plugin)])
    .build();
  let mut runner = SkribbleRunner::new(config);
  let runner_config = runner.initialize()?;
  let mut classes = Classes::default();
  classes.insert_factories(vec![
    ClassFactory::from_tokens(runner_config, &["contained"]),
    ClassFactory::from_tokens(runner_config, &["lg", "contained"]),
  ]);
  classes.sort_by_class();

  insta::assert_display_snapshot!(classes.to_skribble_css(runner_config)?);

  Ok(())
}
