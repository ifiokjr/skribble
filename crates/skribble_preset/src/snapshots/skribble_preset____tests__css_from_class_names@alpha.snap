---
source: crates/skribble_preset/src/__tests.rs
expression: classes.to_skribble_css(runner_config)?
---
/* Generated by Skribble */
@charset "utf-8";
@layer base, alias, default, priority-class;
@property --sk-s-hue {
  syntax: "<number> | <angle>";
  inherits: true;
  initial-value: 314;
}
@property --sk-s-saturation {
  syntax: "<percentage>";
  inherits: true;
  initial-value: 100%;
}
@property --sk-s-lightness {
  syntax: "<percentage>";
  inherits: true;
  initial-value: 47.058823%;
}
@property --sk-s-alpha {
  syntax: "<number> | <percentage>";
  inherits: true;
  initial-value: 1;
}
@property --sk-s {
  syntax: "<color>";
  inherits: true;
  initial-value: hsl(314 100% 47.058823%);
}
@property --sk-p-hue {
  syntax: "<number> | <angle>";
  inherits: true;
  initial-value: 258.89362;
}
@property --sk-p-saturation {
  syntax: "<percentage>";
  inherits: true;
  initial-value: 94.37751%;
}
@property --sk-p-lightness {
  syntax: "<percentage>";
  inherits: true;
  initial-value: 51.17647%;
}
@property --sk-p-alpha {
  syntax: "<number> | <percentage>";
  inherits: true;
  initial-value: 1;
}
@property --sk-p {
  syntax: "<color>";
  inherits: true;
  initial-value: hsl(258.89362 94.37751% 51.17647%);
}
@layer default {
  :root {
    --sk-s-hue: 314;
    --sk-s-saturation: 100%;
    --sk-s-lightness: 47.058823%;
    --sk-s-alpha: 1;
    --sk-s: hsl(314 100% 47.058823%);
    --sk-p-hue: 258.89362;
    --sk-p-saturation: 94.37751%;
    --sk-p-lightness: 51.17647%;
    --sk-p-alpha: 1;
    --sk-p: hsl(258.89362 94.37751% 51.17647%);
  }
  .\(alpha\=0\.05\)\:bg\:\$pink900 {
    background-color: hsl(335.88785 69.032265% 30.392159% / 0.05);
  }
  .\(alpha\=0\.5\)\:bg\:\$secondary {
    background-color: hsl(var(--sk-s-hue) var(--sk-s-saturation) var(--sk-s-lightness) / 0.5);
  }
  .\(alpha\=\=005\)\:bg\:\$red100 {
    background-color: hsl(0 93.333336% 94.117645% / 0.05);
  }
  .\(alpha\=\=050\)\:bg\:\$primary {
    background-color: hsl(var(--sk-p-hue) var(--sk-p-saturation) var(--sk-p-lightness) / 0.5);
  }
}

